### PR TITLE
cegui: drop unused dependencies in makedepends; SMC/cegui07: remove

### PR DESCRIPTION
**Pull request by TinfoilSubmarine**
#### Testing the changes
- I tested the changes in this PR: **YES** (ran `TSC` with updated `gecui`)

Another one not actually using `python-devel`. Configure output shows that it wasn't even building the python extension modules due to some issue with Boost-python. The only package depending on this is `TSC` and it doesn't seem to require the python extensions.

### CODE REVIEW

**Comment by classabbyamp** [comment]
re: the conflict found in CI, might be good to deal eith cegui07's python in this PR too

btw: thanks for taking care of these!

**Comment by classabbyamp** [comment]
hm, the only thing using cegui07 is what TSC is forked from, which hasn't had a commit in 8 years

File: srcpkgs/cegui/template [hunk 1]
```diff
@@ -14,7 +14,6 @@ makedepends="libglvnd-devel SDL2-devel SDL2_image-devel libxml2-devel boost-deve
  devil-devel glfw-devel glm ois-devel silly-devel fribidi-devel lua53-devel minizip-devel
  $(vopt_if irrlicht irrlicht-devel)
  $(vopt_if ogre libogre-devel)
- $(vopt_if python python-devel)
  $(vopt_if gtk gtk+-devel)"
```
Comment (paper42) [inline srcpkgs/cegui/template:17]: what would dropping gtk2 mean?

Comment (TinfoilSubmarine) [reply srcpkgs/cegui/template:17]: http://static.cegui.org.uk/docs/0.8.7/dependencies.html:

>  GTK2 (http://www.gtk.org/)
>  Required to enable the GTK2 based renderer selector used in the samples.

So, probably fine

Comment (paper42) [reply srcpkgs/cegui/template:17]: ok, let's drop it then

**Comment by classabbyamp** [comment]
re: conflict

I'm inclined to say that SMC and cegui07 should be removed. SMC was added in 2015 and hasn't had an update since 2009. Its fork, TSC, has more active development (latest release 2020).

File: srcpkgs/cegui07/template [hunk 2]
```diff
@@ -21,18 +20,10 @@ homepage="http://cegui.org.uk/"
...
-	# XXX: boost-python is not available when cross building
-	build_options_default+=" python"
-fi
-
-# Make configure find python and freetype2
-CFLAGS="-I${XBPS_CROSS_BASE}/usr/include/python2.7"
-
 pre_configure() {
 	# Silence doxygen
 	sed -i doc/doxygen/Makefile.am \
```
Comment (paper42) [inline srcpkgs/cegui07/template:29]: convert all seds to vseds

**Review by classabbyamp — APPROVED** [review]

**Pull request merged**